parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -860,6 +860,10 @@ public void endColumn() throws IOException {
    * @throws IOException if there is an error while writing
    */
   public void endBlock() throws IOException {
+    if (currentRecordCount == 0) {
+      throw new IOException("End block with zero record");
```

Comment:
I would suggest using a `ParquetEncodingException` instead.

Reply:
SGTM.

Reply:
The empty parquet files was used in Drill earlier and now this change breaks the possibility to store empty tables (with header schema only) in parquet files. Any suggestions to bypass this and integrate Parquet-1.12.0 to Drill?

Reply:
@vdiravka, you might want to create a separate jira about this topic so we can discuss it in a more open way. Please, also describe what the empty parquet files are used for.

Reply:
@gszadovszky Done
https://issues.apache.org/jira/browse/PARQUET-2026